Modules/Daq/src/DaqTask.cxx

```diff
@@ -15,140 +15,249 @@
+    mSubSystemsRdhSizes[system.first] = new TH1F(name.c_str(), title.c_str(), 128, 0, 2047);
+    mSubSystemsRdhSizes[system.first]->SetCanExtend(TH1::kXaxis);
+    getObjectsManager()->startPublishing(mSubSystemsRdhSizes[system.first]);
+  }
 }
 
-void DaqTask::startOfActivity(Activity& /*activity*/)
+void DaqTask::startOfActivity(Activity& activity)
 {
-  QcInfoLogger::GetInstance() << "startOfActivity" << AliceO2::InfoLogger::InfoLogger::endm;
-  mPayloadSize->Reset();
-  mNumberSubblocks->Reset();
-  mSubPayloadSize->Reset();
-  mIds->Set(0);
-  mNPoints = 0;
+  ILOG(Info, Support) << "startOfActivity: " << activity.mId << ENDM;
+  reset(); // TODO make sure this is what we want actually, but most probably it is
 }
 
-void DaqTask::startOfCycle() { QcInfoLogger::GetInstance() << "startOfCycle" << AliceO2::InfoLogger::InfoLogger::endm; }
+void DaqTask::startOfCycle()
+{
+  ILOG(Info, Support) << "startOfCycle" << ENDM;
+}
 
 void DaqTask::monitorData(o2::framework::ProcessingContext& ctx)
 {
-  // what does it mean to have several inputs ? is it that we defined several in the config file ?
-  // If I am connected to the readout can I ever receive several inputs ?
-  // TODO maybe we need a special daq task for the readout because we have to look into the weird data structure.
+  monitorInputRecord(ctx.inputs());
+  monitorRDHs(ctx.inputs());
+}
+
+void DaqTask::endOfCycle()
+{
+  ILOG(Info, Support) << "endOfCycle" << ENDM;
+}
+
+void DaqTask::endOfActivity(Activity& /*activity*/)
+{
+  ILOG(Info, Support) << "endOfActivity" << ENDM;
+}
+
+void DaqTask::reset()
+{
+  ILOG(Info, Support) << "Reset" << ENDM;
+
+  // TODO if the number of plots grows we should probably have a container with pointers/references to all of them.
+  //      then we can just iterate over.
 
-  // in a loop
+  mInputRecordPayloadSize->Reset();
+  mNumberInputs->Reset();
+  mInputSize->Reset();
+  mNumberRDHs->Reset();
+
+  for (const auto& system : mSystems) {
+    mSubSystemsRdhSizes.at(system.first)->Reset();
+    mSubSystemsTotalSizes.at(system.first)->Reset();
+  }
+}
+
+void DaqTask::printInputPayload(const header::DataHeader* header, const char* payload)
+{
+  std::vector<std::string> representation;
+  if (mCustomParameters["printInputPayload"] == "hex") {
+    representation = getHexRepresentation((unsigned char*)payload, header->payloadSize);
+  } else if (mCustomParameters["printInputPayload"] == "bin") {
+    representation = getBinRepresentation((unsigned char*)payload, header->payloadSize);
+  }
+
+  for (size_t i = 0; i < representation.size();) {
+    ILOG(Info) << std::setw(4) << i << " : ";
+    for (size_t col = 0; col < 4; col++) {
+      for (size_t word = 0; word < 2; word++) {
+        if (i + col * 2 + word < representation.size()) {
+          ILOG(Info) << representation[i + col * 2 + word];
+        } else {
+          ILOG(Info) << "   ";
+        }
+      }
+      ILOG(Info) << " | ";
+    }
+    ILOG(Info) << ENDM;
```

Comment:
Here we print with severity `Support`, but the page is printed with `Ops`. Is there a reason to make it inconsistent?